lib/src/bootloader.rs

```diff
@@ -12,15 +16,44 @@ pub(crate) fn install_via_bootupd(
     device: &Utf8Path,
     rootfs: &Utf8Path,
     configopts: &crate::install::InstallConfigOpts,
+    via_loopback: bool,
 ) -> Result<()> {
+    let device_str: &str;
+    let prepboot_dir;
     let verbose = std::env::var_os("BOOTC_BOOTLOADER_DEBUG").map(|_| "-vvvv");
     // bootc defaults to only targeting the platform boot method.
     let bootupd_opts = (!configopts.generic_image).then_some(["--update-firmware", "--auto"]);
+
+    if cfg!(target_arch = "powerpc64") {
+        // get PowerPC-PReP-boot device information
+        if via_loopback {
+            // While writing to a qcow2 image use first partition
+            // of loopback device as it points to
+            // PowerPC-PReP-boot partition
+            device_str = "/dev/loop0p1";
+        } else {
+            // While writing to a disk use realpath to extract
+            // PowerPC-PReP-boot partition
+            let result = Command::new("realpath")
+                .arg("-L")
+                .arg("/dev/disk/by-partlabel/PowerPC-PReP-boot")
```

Comment:
This is a complex topic because in the fully general case, there might already be an existing partition with this label on the system. Think about a scenario where we are trying to create a raw disk image from an already booted host.

Now, we're effectively bypassing that here because we special case the loopback path.

Backing up to a higher level here, I've been trying to drive alignment between all the different install paths we have in the Fedora-derivative ecosystem. One of those is Fedora CoreOS and derivatives, see https://github.com/coreos/coreos-assembler/blob/c426a1c34a745762a9314872212e447b9fd26488/src/osbuild-manifests/coreos.osbuild.ppc64le.mpp.yaml#L343 on this topic.

That's basically the loopback install path (generic disk image generation).

Whereas this one should more align with Anaconda...which, I suspect probably needs similar logic with bootupd.

All of this perhaps argues that we should change bootupd to handle this instead?

Reply:
Right, fair point. 

Infact my first preference was lsblk command, not realpath. Since lsblk command takes a device as input we should be able to solve this issue of multiple devices (containing PReP boot) present on a system.  For some unknown reason the command did not work as expected, although the same works outside of bootc.

$ lsblk --paths --noheadings -o name --filter 'PARTLABEL=="PowerPC-PReP-boot"' /dev/sdb
/dev/sdb1

I probably did not codify the logic correctly. --filter option wasn't being enforced correctly and as a result all the partitions on the device were listed in the output. I will revisit the logic again.

Reply:
I think it may be simpler to use our internal wrapper for `lsblk` and then just traverse the children looking for ones with that label in Rust code instead of filtering at the lsblk level.

But I don't object either to adding support for filtering to our lsblk wrapper either.

Can you post (whether as a diff here or just force push to this PR) the code you were trying that wasn't working?

Reply:
I have pushed the latest code.